x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_router.tsx

```diff
@@ -69,29 +69,31 @@ export const EngineRouter: React.FC = () => {
     },
   } = useValues(AppLogic);
 
-  const { dataLoading, engineNotFound } = useValues(EngineLogic);
+  const { engineName, dataLoading, engineNotFound } = useValues(EngineLogic);
   const { setEngineName, initializeEngine, clearEngine } = useActions(EngineLogic);
 
-  const { engineName } = useParams() as { engineName: string };
-  const engineBreadcrumb = [ENGINES_TITLE, engineName];
+  const { engineName: engineNameParam } = useParams() as { engineName: string };
+  const engineBreadcrumb = [ENGINES_TITLE, engineNameParam];
+
+  const isEngineInStateStale = () => engineName !== engineNameParam;
```

Comment:
I think I mentioned this in Slack last year (HEHEHE) but just in case, because last year seems like a blur - it might be great to rename this so it doesn't sound negative/like a bug, but is instead an anticipated load/use case. So something like `isChangingEngines` or `isLoadingNewEngine`.

Reply:
Also looking back again this approach is seriously awesome, I love it!

Reply:
Can you live with it as-is? (Just trying to avoid waiting for another round of CI).

Reply:
Ya for sure! Would you be cool with me opening another follow-up PR later? I don't mind babysitting CI since I'm probably not doing anything big this week

Reply:
Go for it!